cmd/mt-index-analyzer/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/grafana/metrictank/conf"
+	"github.com/grafana/metrictank/idx/cassandra"
+	"github.com/grafana/metrictank/idx/memory"
+	"github.com/grafana/metrictank/logger"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	maxPart      = 512
+	aggregations conf.Aggregations
+	indexRules   conf.IndexRules
+	schemas      conf.Schemas
+
+	aggFile        = "/etc/metrictank/storage-aggregation.conf"
+	indexRulesFile = "/etc/metrictank/index-rules.conf"
+	schemasFile    = "/etc/metrictank/storage-schemas.conf"
+
+	totalPointsInTank  uint64
+	totalPointsInStore uint64
+	totalSeries        uint64
+
+	dist    sync.Mutex
+	irDist  []int
+	aggDist []int
+)
+
+func init() {
+	formatter := &logger.TextFormatter{}
+	formatter.TimestampFormat = "2006-01-02 15:04:05.000"
+	log.SetFormatter(formatter)
+	log.SetLevel(log.InfoLevel)
+}
+
+func perror(err error) {
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+func main() {
+	fl := cassandra.ConfigSetup()
+	fl.StringVar(&aggFile, "aggregations-file", aggFile, "path to storage-aggregation.conf file")
+	fl.StringVar(&indexRulesFile, "rules-file", indexRulesFile, "path to index-rules.conf file")
+	fl.StringVar(&schemasFile, "schemas-file", schemasFile, "path to storage-schemas.conf file")
+	fl.IntVar(&maxPart, "max-partition", maxPart, "id of the highest partition to look for")
+
+	flag.Usage = func() {
+		fmt.Println("mt-index-analyzer")
+		fmt.Println("")
+		fmt.Println("analyzes the contents of your index")
+		fmt.Println("limitations:")
+		fmt.Println(" - does not take into account series churn. more specifically, assumes each series has data for its entire defined retention")
+		fmt.Println(" - assumes chunks are filled corresponding to the interval, doesn't account for sparse data or too-high resolution data")
+		fl.PrintDefaults()
+	}
+	if len(os.Args) == 2 && (os.Args[1] == "-h" || os.Args[1] == "--help") {
+		flag.Usage()
+		os.Exit(0)
+	}
+
+	fl.Parse(os.Args[1:])
+	cassandra.CliConfig.Enabled = true
+	cassandra.ConfigProcess()
+	idx := cassandra.New(cassandra.CliConfig)
+	err := idx.InitBare()
+	perror(err)
+
+	// for loading the index, to not drop any entries
+	memory.IndexRules = conf.IndexRules{
+		Rules: nil,
+		Default: conf.IndexRule{
+			Name:     "default",
+			Pattern:  regexp.MustCompile(""),
+			MaxStale: 0,
+		},
+	}
+
+	loadConfFiles()
+
+	irDist = make([]int, len(indexRules.Rules)+1)   // we need one extra to mark the default case
+	aggDist = make([]int, len(aggregations.Data)+1) // we need one extra to mark the default case
+
+	jobs := make(chan Job, 10000)
+
+	go loadJobs(idx, jobs)
+
+	var wg sync.WaitGroup
+	wg.Add(10)
+	for i := 0; i < 10; i++ {
+		go processJobs(&wg, jobs)
+	}
+	wg.Wait()
+
+	fmt.Println("index-rules distribution:", irDist)
```

Comment:
The output needs to be formatted so that it is clear what is being printed. 
eg maybe something like
```
<indexRuleName> : <count>
```